source/configuration_manual/replication.rst

```diff
@@ -0,0 +1,415 @@
+.. _replication:
+
+======================
+Replication with dsync
+======================
+
+.. note::
+
+  This is not supported with Dovecot Pro.
+
+Dovecot supports master/master replication using dsync. It's recommended
+that the same user always gets redirected to the same replica, but no
+changes get lost even if the same user modifies mails simultaneously on
+both replicas, some mails just might have to be redownloaded. The
+replication is done asynchronously, so high latency between the replicas
+isn't a problem. The replication is done by looking at Dovecot index
+files (not what exists in filesystem), so no mails get lost due to
+filesystem corruption or an accidental ``rm -rf``, they will simply be
+replicated back.
+
+Replication works only between server pairs. If you have a large
+cluster, you need multiple independently functioning Dovecot backend
+pairs. This means that :ref:`director <dovecot_director>` isn't
+currently supported with replication. The replication in general is a
+bit resource intensive, so it's not recommended to be used in
+multi-million user installations.
+
+.. warning::
+
+  Shared folder replication doesn't work correctly right now.
```

Comment:
I would consider removing "right now" here, in case we have concrete plans to fix this. The section about possible locking strategies later in this paragraph sounds a more like design than necessary technical documentation, but maybe it helps to explain the issue.